ext/ddtrace.c

```diff
@@ -651,6 +651,35 @@ PHP_METHOD(DDTrace_SpanLink, jsonSerialize) {
     RETURN_ARR(array);
 }
 
+/* DDTrace\SpanEvent */
+zend_class_entry *ddtrace_ce_span_event;
+
+PHP_METHOD(DDTrace_SpanEvent, jsonSerialize) {
+    ddtrace_span_event *event = (ddtrace_span_event*)Z_OBJ_P(ZEND_THIS);
+
+    zend_array *array = zend_new_array(3);
+
+    zend_string *name = zend_string_init("name", sizeof("name") - 1, 0);
+    zend_string *timestamp = zend_string_init("time_unix_nano", sizeof("time_unix_nano") - 1, 0);
+
+    Z_TRY_ADDREF(event->property_name);
+    zend_hash_add(array, name, &event->property_name);
```

Comment:
You can use `zend_hash_str_add(array, ZEND_STRL("name"), &event->property_name)` instead of allocating zend_string manually.